lib/rubocop/cop/lint/useless_default_value_argument.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for usage of method `fetch` or `Array.new` with default value argument
+      # and block. In such cases, block will always be used as default value.
+      #
+      # This cop emulates Ruby warning "block supersedes default value argument" which
+      # applies to `Array.new`, `Array#fetch`, `Hash#fetch`, `ENV.fetch` and
+      # `Thread#fetch`.
+      #
+      # @safety
+      #   This cop is unsafe because the receiver could have nonstandard implementation
+      #   of `fetch`, or be a class other than the one listed above.
+      #
+      #   It is also unsafe because default value argument could have side effects:
+      #   [source,ruby]
+      #   ----
+      #   def x(a) = puts "side effect"
+      #   Array.new(5, x(1)) { 2 }
+      #   ----
+      #   so removing it would change behavior.
```

Comment:
```suggestion
      #   It is also unsafe because default value argument could have side effects:
      #
      #   [source,ruby]
      #   ----
      #   def x(a) = puts "side effect"
      #   Array.new(5, x(1)) { 2 }
      #   ----
      #
      #   so removing it would change behavior.
```

Reply:
Updated